src/impl_methods.rs

```diff
@@ -1281,8 +1281,7 @@ where
 
     /// Return true if the array is known to be contiguous.
     ///
-    /// Will detect c- and f-contig arrays correctly, but otherwise
-    /// There are some false negatives.
+    /// Will detect c- and f-contig arrays correctly.
```

Comment:
Just checking, isn't it supposed to detect all contiguous arrays now correctly? Could there be some test cases for that?

I mean like this:

Start with C- or F- contig array.

Swap any axes. Invert any axes.

The result should always be contiguous, right? If there are cases we still don't detect, it would be great to clarify what that is (and keep the "disclaimer" in the doc).

Reply:
I did the following test, no matter the layout is C, Fortran or custom, it got the correct result. I tried to put these tests into pr, but because is_contiguous is not a public API (it is pub(crate) ), it is not possible to add tests directly.
    use ndarray::ArrayView;
    use ndarray::Axis;
    use ndarray::arr3;
    use ndarray::arr2;
    use ndarray::ShapeBuilder;
    use ndarray::s;
    fn main() {
        test3()
    }
    fn test2(){
        let s = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13];
        let mut a = ArrayView::from_shape((2, 3, 2).strides((1, 4, 2)),
                                      &s).unwrap();
        assert_eq!(a, arr3(&[[[0, 2], [4, 6], [8, 10]], [[1, 3], [5, 7], [9, 11]]]));
        assert!(a.is_contiguous());

        let mut b=a.slice(s![..;1, ..;-1, ..;-1]);
        assert_eq!(b, arr3(&[[[10, 8], [6, 4], [2, 0]], [[11, 9], [7, 5], [3, 1]]]));
        assert!(b.is_contiguous());

        b.swap_axes(1,2);
        assert_eq!(b, arr3(&[[[10, 6, 2], [8, 4, 0]], [[11, 7, 3], [9, 5, 1]]]));
        assert!(b.is_contiguous());

        b.invert_axis(Axis(0));
        assert_eq!(b, arr3(&[[[11, 7, 3], [9, 5, 1]], [[10, 6, 2], [8, 4, 0]]]));
        assert!(b.is_contiguous());

        let mut c=b.reversed_axes();
        assert_eq!(c, arr3(&[[[11, 10],[9, 8]],[[7, 6],[5, 4]],[[3, 2],[1, 0]]]));
        assert!(c.is_contiguous());

        c.merge_axes(Axis(1),Axis(2));
        assert_eq!(c, arr3(&[[[11, 10, 9, 8]], [[7, 6, 5, 4]], [[3, 2, 1, 0]]]));
        assert!(c.is_contiguous());

        let d=b.remove_axis(Axis(1));
        assert_eq!(d, arr2(&[[11, 7, 3], [10, 6, 2]]));
        assert!(!d.is_contiguous());

        let e=b.remove_axis(Axis(2));
        assert_eq!(e, arr2(&[[11, 9],[10, 8]]));
        assert!(e.is_contiguous());

        let f=e.insert_axis(Axis(2));
        assert_eq!(f, arr3(&[[[11], [9]], [[10], [8]]]));
        assert!(f.is_contiguous());

        let mut g=b.clone();
        g.collapse_axis(Axis(1),0);
        assert_eq!(g, arr3(&[[[11, 7, 3]],[[10, 6, 2]]]));
        assert!(!g.is_contiguous());

        b.collapse_axis(Axis(2),0);
        assert_eq!(b, arr3(&[[[11], [9]], [[10], [8]]]));
        assert!(b.is_contiguous());

        let h = ArrayView::from_shape((1, 3, 2).strides((2, 4, 2)),
                                      &s).unwrap();
        assert_eq!(h, arr3(&[[[0, 2], [4, 6], [8, 10]]]));
        assert!(!h.is_contiguous());
    }